boolean/boolean.py

```diff
@@ -1032,55 +1094,61 @@ def start_operation(ast, operation):
                 ast[0].append(ast[1](*ast[2:], eval=eval))
                 ast = ast[0]
 
-    expr = expr.replace(" ", "")
-    length = len(expr)
+    if isinstance(expr, str):
+        tokenized = tokenizer(expr)
+    else:
+        try:
+            tokenized = list(expr)
+        except:
```

Comment:
the wrapping in a list is a remnant of the original while loop that needed a list.
I would think that just accepting an iterable is enough. A callable  would be a bit too much to except, don't you think?
And I would remove the try/except entirely and let the exception bubble up if any.